WcaOnRails/app/models/competition.rb

```diff
@@ -742,23 +742,21 @@ def self.search(query, params: {})
       competitions = competitions.where("CAST(CONCAT(year,'-',endMonth,'-',endDay) as Date) <= ?", end_date)
     end
 
-    if query.present?
-      sql_query = "%#{query}%"
-      competitions = competitions.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    query&.split&.each do |part|
+      like_query = %w(id name cellName cityName countryId).map { |column| column + " LIKE :part" }.join(" OR ")
+      competitions = competitions.where(like_query, part: "%#{part}%")
     end
 
-    if params[:sort].present?
-      params[:sort].split(",").each do |field|
-        order = field.start_with?("-") ? :desc : :asc
+    (params[:sort] || "-start_date").split(",").each do |field|
```

Comment:
@jonatanklosko and I just discussed this offline, and he pointed out that in this particular case, we did not break AFS as AFS was requesting decreasing start date, and that's all we support right now.

I personally like the comma separated list of fields to sort by with - for decreasing (I did not invent this myself, I did a bit of reading on APIs before picking it), but I'm happy to wait until we're ready to document our api before having this discussion.